met-web/src/components/engagement/form/EngagementWidgets/Documents/FileDrawer.tsx

```diff
@@ -42,25 +45,63 @@ const FileDrawer = () => {
 
     const { handleSubmit, reset } = methods;
 
+    useEffect(() => {
+        methods.setValue('name', documentToEdit?.title || '');
+        methods.setValue('link', documentToEdit?.url || '');
+        methods.setValue('folderId', documentToEdit?.parent_document_id || 0);
+    }, [documentToEdit]);
+
     const handleClose = () => {
         reset();
         handleFileDrawerOpen(false);
     };
 
+    const updateDocument = async (data: FileForm) => {
+        if (!(documentToEdit && widget)) {
+            return;
+        }
+        const documentEditsToPatch = updatedDiff(documentToEdit, {
+            title: data.name,
+            parent_document_id: data.folderId === 0 ? null : data.folderId,
+            url: data.link,
+        }) as PatchDocumentRequest;
+        await patchDocument(widget.id, documentToEdit.id, {
+            ...documentEditsToPatch,
+            parent_document_id: data.folderId === 0 ? null : data.folderId,
+        });
+    };
+
+    const createDocument = async (data: FileForm) => {
+        if (!widget) {
+            return;
+        }
+        await postDocument(widget.id, {
+            title: data.name,
+            parent_document_id: data.folderId === 0 ? null : data.folderId,
+            url: data.link,
+            widget_id: widget.id,
+            type: 'file',
+        });
+    };
+
     const onSubmit: SubmitHandler<FileForm> = async (data: FileForm) => {
         if (!widget) {
             return;
         }
         try {
             setIsCreatingDocument(true);
-            await postDocument(widget.id, {
-                title: data.name,
-                parent_document_id: data.folderId === 0 ? null : data.folderId,
-                url: data.link,
-                widget_id: widget.id,
-                type: 'file',
-            });
-            await loadDocuments();
+            if (documentToEdit) {
+                updateDocument(data);
```

Comment:
separate this into another method called saveDocument, have it like the example I shared earlier